intern/src/lib.rs

```diff
@@ -9,29 +9,21 @@ use std::ptr::NonNull;
 use std::sync::Arc;
 
 pub trait InternString {
-    unsafe fn to_interned(s: ArenaStr) -> Self;
+    unsafe fn to_interned(s: InternedArenaStr) -> Self;
```

Comment:
I know this wasn't documented before, but can we add documentation on the safety requirements for this `unsafe` function?